src/com/google/enterprise/adaptor/DocIdSender.java

```diff
@@ -330,7 +330,7 @@ private String alternateGroupSource(String source) {
       log.log(Level.WARNING,
           "GSA ver {0} does not accept group definitions", gsaVerString);
       journal.recordGroupPushFailed();
-      return defs.entrySet().iterator().next().getKey();
+      return defs.isEmpty() ? null : defs.entrySet().iterator().next().getKey();
```

Comment:
Add tests for these two bugs. Maybe change this to defs.keySet().iterator().next(), since we're changing this line anyway?